buildmodel/commands.go

```diff
@@ -319,6 +318,20 @@ func SubmitUpdatePR(f *PRFlags) error {
 	return nil
 }
 
+// GetTempPathInDir creates a temporary path inside the given root dir. Turns local time into a
+// sortable path-friendly name to use as the name. This allows a command to run multiple times in
+// sequence without overwriting or deleting the old data, for diagnostic purposes. This is intended
+// as a dev convenience, not to enable running in parallel.
+func GetTempPathInDir(rootDir string) (string, error) {
+	pathDate := time.Now().Format("2006-01-02_15-04-05.000")
+	p := path.Join(rootDir, pathDate)
+
+	if _, err := os.Stat(p); !errors.Is(err, os.ErrNotExist) {
+		return "", fmt.Errorf("temporary Git dir already exists: %v", p)
```

Comment:
nitpick: it is usually a good practice to format an error using the `%w` verb so the new errors wraps the original one. This way the original error can still be retrieved in case it is necessary later on.

Reply:
Hmm, I actually think this code has a bug in it, seeing it again. If there's an `err` returned that isn't `os.ErrNotExist`, this function will ignore it completely. I'm not sure what error that would be... but better to catch weird state here than let it affect something later on. It should be handled more like this, I think:

```go
	// Use Stat to make sure this path doesn't exist. Ignore result and check for correct error.
	if _, err := os.Stat(p); err != nil {
		if errors.Is(err, os.ErrNotExist) {
			// Good: the dir should not exist.
			return p, nil
		} else {
			return "", fmt.Errorf("unknown error while checking if temporary Git dir '%v' exists: %w", p, err)
		}
	}
	return "", fmt.Errorf("temporary Git dir already exists: %v", p)
```

(Flipping both of the conditions would make it less nested and keep success at the bottom of the func, but I think leaving it with the usual `!=` and positive `errors.Is` check probably makes it more readable on first glance.)